File: DSAlgorithms/Leetcode/c++/arrays_hash/523-continuous_subarr_sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
       std::unordered_map<int, int> hash = {{0, -1}};
       int total = 0;
       for(int i = 0; i < nums.size(); i++)
       {
           total += nums[i];
           int remainder = total % k;
           if(!(hash.count(remainder) > 0))
           {
               hash[remainder] = i;
           }
           else if(i - hash[remainder] > 1)
                return true;
       }
        return false;

    }
};

File: DSAlgorithms/Leetcode/ts/src/arrays_hash/523-continuous_subarr_sum.ts
const checkSubarraySum = (nums: number[], k: number): boolean =>{
    const hash = new Map<number, number> ();
    hash.set(0, -1);
    let total: number = 0;
    for(let i = 0; i < nums.length; i++)
    {
        total += nums[i];
        let remainder: number = total % k;
        if(!hash.has(remainder))
            hash.set(remainder, i);
        else if((i - hash.get(remainder)) > 1)
            return true;
    }
    return false;

};